src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -941,10 +941,945 @@ class spell_silverpine_forsaken_trooper_masterscript_high_command : public Spell
+{
+    npc_silverpine_lord_darius_crowley_exsanguinate(Creature* creature) : ScriptedAI(creature) { }
+
+    void Reset() override
+    {
+        me->SetReactState(REACT_PASSIVE);
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (Player* player = summoner->ToPlayer())
+            _playerGUID = player->GetGUID();
+
+        FindAllGuid();
+    }
+
+    void WaypointReached(uint32 waypointId, uint32 pathId) override
+    {
+        if (pathId == PATH_DARIUS_EXIT && waypointId == WAYPOINT_ON_DARIUS_DESPAWN)
+            me->DespawnOrUnsummon();
+    }
+
+    void FindAllGuid()
+    {
+        if (!_bloodfangGUID)
+        {
+            if (Creature* ivar = me->FindNearestCreature(NPC_PACKLEADER_IVAR_BLOODFANG, 100.0f))
+                _bloodfangGUID = ivar->GetGUID();
+        }
+
+        if (!_armoireGUID)
+        {
+            if (Creature* armoire = me->FindNearestCreature(NPC_ARMOIRE_SUMMONED, 100.0f))
+            {
+                _armoireGUID = armoire->GetGUID();
+
+                if (armoire->IsAIEnabled())
+                    armoire->GetAI()->SetGUID(me->GetGUID(), me->GetEntry());
+            }
+        }
+    }
+
+private:
+    EventMap _events;
+    ObjectGuid _playerGUID;
+    ObjectGuid _armoireGUID;
+    ObjectGuid _bloodfangGUID;
+};
+
+enum IvarBloodfangExsanguinate
+{
+    WAYPOINT_ON_IVAR_DESPAWN                = 3
+};
+
+// Packleader Ivar Bloodfang - 44884
+struct npc_silverpine_packleader_ivar_bloodfang_exsanguinate : public ScriptedAI
+{
+    npc_silverpine_packleader_ivar_bloodfang_exsanguinate(Creature* creature) : ScriptedAI(creature) { }
+
+    void Reset() override
+    {
+        me->SetReactState(REACT_PASSIVE);
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        if (Player* player = summoner->ToPlayer())
+            _playerGUID = player->GetGUID();
+
+        if (Creature* armoire = me->FindNearestCreature(NPC_ARMOIRE_SUMMONED, 30.0f))
+        {
+            _armoireGUID = armoire->GetGUID();
+
+            if (armoire->IsAIEnabled())
+                armoire->GetAI()->SetGUID(me->GetGUID(), me->GetEntry());
+        }
+    }
+
+    void WaypointReached(uint32 waypointId, uint32 pathId) override
+    {
+        if (pathId == PATH_IVAR_EXIT && waypointId == WAYPOINT_ON_IVAR_DESPAWN)
+            me->DespawnOrUnsummon();
+    }
+
+private:
+    EventMap _events;
+    ObjectGuid _playerGUID;
+    ObjectGuid _armoireGUID;
+};
+
+enum EjectPassenger1
+{
+    SEAT_BLOODFANG                          = 0
+};
+
+// Eject Passenger 1 - 80743
+class spell_silverpine_eject_passenger_1 : public SpellScript
+{
+    PrepareSpellScript(spell_silverpine_eject_passenger_1);
+
+    void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+    {
+        if (GetHitUnit()->IsVehicle())
+        {
+            if (Unit* passenger0 = GetHitUnit()->ToUnit()->GetVehicleKit()->GetPassenger(SEAT_BLOODFANG))
```

Comment:
`Unit* SpellScript::GetHitUnit() const`
-> `ToUnit()` is unnecessary

Reply:
Feedback applied!